pkg/services/login/authinfoservice/service.go

```diff
@@ -0,0 +1,221 @@
+package authinfoservice
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/login"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+const genericOAuthModule = "oauth_generic_oauth"
+
+func init() {
+	srv := &Implementation{}
+
+	registry.Register(&registry.Descriptor{
+		Name:         "UserAuthInfo",
+		Instance:     srv,
+		InitPriority: registry.MediumHigh,
+	})
+}
+
+type Implementation struct {
+	Bus                   bus.Bus                     `inject:""`
+	SQLStore              *sqlstore.SQLStore          `inject:""`
+	UserProtectionService login.UserProtectionService `inject:""`
+
+	logger log.Logger
+}
+
+func (s *Implementation) Init() error {
+	s.logger = log.New("login.authinfo")
+
+	s.Bus.AddHandler(s.GetExternalUserInfoByLogin)
```

Comment:
Might look into removing these bus handlers and adding them to the service interface instead if possible.

Reply:
Skipping that for this PR, better to do that another time and to get this merged now,.